lib/ansible_tower_client/job_template.rb

```diff
@@ -27,5 +34,28 @@ def survey_spec
     def self.endpoint
       "job_templates".freeze
     end
+
+    private
+
+    def with_temporary_changes(limit)
+      old_limit = @raw_body['limit']
+      new_limit = limit
+      patch("{ \"limit\": \"#{new_limit}\" }")
+      begin
+        yield
+      rescue => e
```

Comment:
Don't rescue all exceptions.  I expect that any errors would be from the `api.post` [here](https://github.com/ManageIQ/ansible_tower_client/pull/22/files#diff-c799bcb40bc98d6ec3bcd2e4b41361e6R17) and only those should be rescued.